File: Javascript/Group-Anagrams.js
function groupAnagrams(strs) {
  let sortedStrings = {};

  strs.forEach((element, index) => {
    let sortedString = element.split("").sort().join("");
    if (sortedStrings.hasOwnProperty(sortedString)) {
      sortedStrings[sortedString].push(element);
    } else {
      sortedStrings[sortedString] = [element];
    }
  });

  return Object.values(sortedStrings);
}

function groupAnagrams(strs) {
  let sortedStrings = {};

  strs.forEach((str) => {
    let sortedString = str.split("").sort().join("");

    if (sortedString in sortedStrings) {
      sortedStrings[sortedString].push(str);
    } else {
      sortedStrings[sortedString] = [str];
    }
  });

  return Object.values(sortedStrings);
}

File: Dart/Group-Anagrams.dart
class Solution {
  List<List<String>> groupAnagrams(List<String> strs) {
    var map = Map<String, List<String>>();
    final a = "a".codeUnitAt(0);

    for (String str in strs) {
      var count = List.filled(26, 0);
      for (int i = 0; i < str.length; i++) {
        count[str[i].codeUnitAt(0) - a]++;
      }

      var key = count.join("#");
      if (map.containsKey(key)) {
        map[key]!.add(str);
        continue;
      }

      map[key] = [str];
    }

    return List.from(map.values);
  }
}
